### ISSUE TITLE
[13.0] Clicking a product's "On Hand" prompts an error after using "Products" breadcrumb

### ISSUE THREAD

**Issue by andrenaught** [odoo/odoo#40537]
Impacted versions: 13.0
Known Impacted Modules: `stock`, `stock_account`

Steps to reproduce:
1. Go to Inventory -> Master Data -> Products
2. Click "Group By" to group the view, choose a grouping (you can just choose "Product Category", but any should work), the view should change.
3. Click "Favorites" -> Save Current Search -> Check "Use by default" -> Save
4. Open a new tab (or refresh). Go to Inventory -> Master Data -> Products
5. Click on a product -> Click on the "On Hand" button -> it should work. We can see that it works properly.
6. Now click on the "Products" breadcrumb (found on the top left) to go back to the Products Master Data. Try to do step 5 again (click on a product -> click on the "On Hand" button). You should get an error.

```
Odoo Server Error
Traceback (most recent call last):
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/http.py", line 619, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/http.py", line 309, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/tools/pycompat.py", line 14, in reraise
    raise value
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/http.py", line 664, in dispatch
    result = self._call_function(**self.params)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/http.py", line 345, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/service/model.py", line 93, in wrapper
    return f(dbname, *args, **kwargs)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/http.py", line 338, in checked_call
    result = self.endpoint(*a, **kw)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/http.py", line 909, in __call__
    return self.method(*args, **kw)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/http.py", line 510, in response_wrap
    response = f(*args, **kw)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/addons/web/controllers/main.py", line 1319, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/addons/web/controllers/main.py", line 1311, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/api.py", line 391, in call_kw
    result = _call_kw_model(method, model, args, kwargs)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/api.py", line 364, in _call_kw_model
    result = method(recs, *args, **kwargs)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/addons/web/models/models.py", line 74, in web_read_group
    expand_limit, expand_orderby)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/addons/web/models/models.py", line 98, in _web_read_group
    orderby=orderby, lazy=lazy)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/addons/stock/models/stock_quant.py", line 157, in read_group
    result = super(StockQuant, self).read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/models.py", line 2123, in read_group
    result = self._read_group_raw(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/models.py", line 2151, in _read_group_raw
    annotated_groupbys = [self._read_group_process_groupby(gb, query) for gb in groupby_list]
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/models.py", line 2151, in <listcomp>
    annotated_groupbys = [self._read_group_process_groupby(gb, query) for gb in groupby_list]
  File "/Users/andrevictoria/Desktop/projects/erp/odoo_test/odoo/models.py", line 1955, in _read_group_process_groupby
    raise ValueError("Invalid field %r on model %r" % (split[0], self._name))
ValueError: Invalid field 'categ_id' on model 'stock.quant'
```

Essentially this happens when you use breadcrumbs within the Inventory module, while having a "Favorited" view that uses the "Group By" organizer.

This is tested on a fresh install (new database, no custom configs, latest odoo on git as of today)

**Current behavior:** Error shows, cannot access "On Hand" view for some reason when using saved views for Product Master Data and using the breadcrumb.

### Where the issue seems to be
It looks like the issue can be traced within the `read_group()` function in the `stock_quant.py` file, found in both the `stock` and `stock_account` modules. I think it's within the `fields` argument being passed to it, or the fact that `read_group()` is being called at all. When I printed out `fields` within this file, like shown:

```python
# addons/stock/models/stock_quant.py
@api.model
def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
    """ This override is done in order for the grouped list view to display the total value of
    the quants inside a location. This doesn't work out of the box because `value` is a computed
    field.
    """
    print (fields)
    if 'value' not in fields:
        return super(StockQuant, self).read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
    res = super(StockQuant, self).read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
```

I saw the difference when I clicked "On Hand" on the frontend.

**When clicking "On Hand" worked**
* `read_group()` doesn't get called. No print out.

**When clicking "On Hand" didn't work**
* `read_group()` is called.
* `fields` printed out:
`['id', 'tracking', 'product_id', 'location_id', 'lot_id', 'package_id', 'owner_id', 'quantity', 'inventory_quantity', 'reserved_quantity', 'product_uom_id', 'company_id', 'categ_id']
`
    * Not sure if it matters but `fields` was printing out `['product_id', 'quantity', 'reserved_quantity']` when I checked the logs before "On Hand" was clicked.

**Comment by Yenthe666** [comment]
Hi @andrenaught,

Thanks a lot for your bug report. I've tested this on the latest Odoo 13.0 test instance and was able to reproduce this bug:
![image](https://user-images.githubusercontent.com/6352350/70392818-ee081880-19e3-11ea-9d11-1c5731fa704c.png)

@sle-odoo maybe somebody from your team can have a look?
Regards,
Yenthe

**Comment by sle-odoo** [comment]
hello,
thanks for the report

to me, it looks more like a webclient issue: you set a default filter on a model A (product.product in this case) and it gets applied to a model B (stock.quant in this case) while it shouldn't. I thinks this is a generic issue due to the way the default filters work and the context is passed in odoo. 

we could erase the `group_by` context key when clicking on the on hand button but what if the presence of this context key was legitimate?

@VincentSchippefilt fyi

**Comment by sle-odoo** [comment]
fixed by 901d60cd90f1e6ba0fd6502c04619193d3ca1c7b
